src/backends/r1cs/snarkjs.rs

```diff
@@ -229,68 +247,75 @@ impl WitnessWriter {
         let current_pos = self.inner.stream_position().unwrap();
         // Calculate the size of the section
         let section_size = current_pos - self.section_size_position - 8;
 
         // Move back to where the size needs to be written
-        self.inner.seek(SeekFrom::Start(self.section_size_position));
+        self.inner
+            .seek(SeekFrom::Start(self.section_size_position))?;
         // Write the actual section size
-        self.inner.write_all(&section_size.to_le_bytes());
+        self.inner.write_all(&section_size.to_le_bytes())?;
         // Return to the end of the section
-        self.inner.seek(SeekFrom::Start(current_pos));
+        self.inner.seek(SeekFrom::Start(current_pos))?;
         // Flush the buffer to ensure all data is written to the file
-        self.inner.flush();
+        self.inner.flush()?;
 
         self.writing_section = None;
+
+        Ok(())
     }
 
     /// Write the witness to the file
     /// It stores the two sections:
     /// - Header section: describes the field size, prime field, and the number of witnesses.
     /// - Witness section: contains the witness values.
-    fn write(&mut self, witness: Vec<BigInt>, prime: &BigInt) {
+    fn write(&mut self, witness: Vec<BigInt>, prime: &BigInt) -> Result<(), io::Error> {
         // Start writing the first section
-        self.start_write_section(1);
+        self.start_write_section(1)?;
         // Write field size in number of bytes
         let field_n_bytes = field_size(prime);
-        self.inner.write_all(&(field_n_bytes as u32).to_le_bytes());
+        self.inner
+            .write_all(&(field_n_bytes as u32).to_le_bytes())?;
         // Write the prime field in bytes
-        self.write_big_int(prime.clone(), field_n_bytes);
+        self.write_big_int(prime.clone(), field_n_bytes)?;
         // Write the number of witnesses
-        self.inner.write_all(&(witness.len() as u32).to_le_bytes());
+        self.inner
+            .write_all(&(witness.len() as u32).to_le_bytes())?;
 
-        self.end_write_section();
+        self.end_write_section()?;
 
         // Start writing the second section
-        self.start_write_section(2);
+        self.start_write_section(2)?;
 
         /// Write the witness values to the file
         /// Each witness value occupies the same number of bytes as the prime field
         for value in witness {
-            self.write_big_int(value.clone(), field_n_bytes as usize);
+            self.write_big_int(value.clone(), field_n_bytes)?;
         }
-        self.end_write_section();
+        self.end_write_section()
     }
 
     /// Write a BigInt to the file
-    fn write_big_int(&mut self, value: BigInt, size: usize) {
+    fn write_big_int(&mut self, value: BigInt, size: usize) -> Result<(), io::Error> {
         let bytes = value.to_bytes_le().1;
 
         let mut buffer = vec![0u8; size];
         buffer[..bytes.len()].copy_from_slice(&bytes);
-        self.inner.write_all(&buffer);
```

Comment:
This can be just:

```
self.inner.write_all(&buffer)
```

Reply:
fixed in [ea2564a](https://github.com/zksecurity/noname/pull/105/commits/ea2564a9a7f3446f2db56f2fbb94733896b32c98)